tools/offline_log_viewer/consumer_offsets.py

```diff
@@ -0,0 +1,118 @@
+import base64
+from io import BytesIO
+from model import *
+from reader import Endianness, Reader
+from storage import Segment
+import datetime
+
+
+def decode_key_type(v):
+    if v == 0 or v == 1:
+        return "offset_commit"
+    elif v == 2:
+        return "group_metadata"
+
+    return "unknown"
+
+
+def decode_member_proto(rdr):
+    ret = {}
+    ret['name'] = rdr.read_string()
+    ret['metadata'] = rdr.read_iobuf().hex()
+    return ret
+
+
+def decode_member(rdr):
+    ret = {}
+    ret['v'] = rdr.read_int16()
+    ret['member_id'] = rdr.read_kafka_string()
+    ret['instance_id'] = rdr.read_kafka_optional_string()
+    ret['client_id'] = rdr.read_kafka_string()
+    ret['client_host'] = rdr.read_kafka_string()
+    ret['rebalance_timeout'] = rdr.read_int32()
+    ret['session_timeout'] = rdr.read_int32()
+    ret['subscription'] = base64.b64encode(
+        rdr.read_kafka_bytes()).decode('utf-8')
+    ret['assignment'] = base64.b64encode(
+        rdr.read_kafka_bytes()).decode('utf-8')
+
+    return ret
+
+
+def decode_metadata(rdr):
+    ret = {}
+    ret['version'] = rdr.read_int16()
+    ret['protocol_type'] = rdr.read_kafka_string()
+    ret['generation_id'] = rdr.read_int32()
+    ret['protocol_name'] = rdr.read_kafka_optional_string()
+    ret['leader'] = rdr.read_kafka_optional_string()
+    ret['state_timestamp'] = rdr.read_int64()
+    ret['member_state'] = rdr.read_vector(decode_member)
+    return ret
+
+
+def decode_key(key_rdr):
+    ret = {}
+    v = key_rdr.read_int16()
+    ret['type'] = decode_key_type(v)
+    ret['group_id'] = key_rdr.read_kafka_string()
+    if ret['type'] == 'offset_commit':
+        ret['topic'] = key_rdr.read_kafka_string()
+        ret['partition'] = key_rdr.read_int32()
+
+    return ret
+
+
+def decode_offset_commit(v_rdr):
+    ret = {}
+    ret['version'] = v_rdr.read_int16()
+    ret['committed_offset'] = v_rdr.read_int64()
+    ret['leader_epoch'] = v_rdr.read_int32()
+    ret['committed_metadata'] = v_rdr.read_kafka_string()
+    ret['timestamp'] = v_rdr.read_int64()
```

Comment:
question, was there one field that we added / removed to the offset_commit struct in a recent PR? Would we need to perform a version check to handle that case?